### PR TITLE
Add pod_infra_container_image support to local-up-cluster.sh script

### PR DESCRIPTION
**Pull request by nkwangleiGIT**
**What this PR does / why we need it**:
This PR will add the support to customize pod_infra_container_image parameter of kubelet using environment variable, so we can easily use the infra image of ourself, it'll help in two cases:
1) The user customize the infra image and need to let kubelet pull from the private docker registry

2) Due to GFW in China, we're not able to pull images from gcr.io, so we need this parameter to use local docker registry service to pull the image

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```
NONE
```

### CODE REVIEW

**Comment by nkwangleiGIT** [comment]
I signed the CLA!

**Comment by nkwangleiGIT** [comment]
@lavalamp this is my first time to submit pull request, let me know if anything I should fix, thanks!

File: hack/local-up-cluster.sh [hunk 1]
```diff
@@ -538,6 +539,11 @@ function start_kubelet {
         image_service_endpoint_args="--image-service-endpoint=${IMAGE_SERVICE_ENDPOINT}"
       fi
 
+      pod_infra_container_image=""
+      if [[ -n "${POD_INFRA_CONTAINER_IMAGE}" ]]; then
+        pod_infra_container_image="--pod_infra_container_image=${POD_INFRA_CONTAINER_IMAGE}"
```
Comment (lavalamp) [inline hack/local-up-cluster.sh]: Please use dashes instead of underscores in our flags. Even though it accepts both, dashes are canonical.

Comment (nkwangleiGIT) [inline hack/local-up-cluster.sh]: Just fixed, thanks for pointing this out!

**Comment by nkwangleiGIT** [comment]
@lavalamp please review again, thanks!

**Comment by nkwangleiGIT** [comment]
About "cla/linuxfoundation" label, not sure what's the left problem of "nkwangleiGIT authorized, but 1 other problem", any suggestion about this? thanks!

**Comment by nkwangleiGIT** [comment]
To fix CLA issue, will submit again using another PR, thanks!

**Pull request closed without merging**